ftpvl/fetchers.py

```diff
@@ -68,16 +74,88 @@ def __init__(
         eval_num: int = 0,
+        absolute_eval_num: bool = False,
         mapping: dict = None,
         hydra_clock_names: list = None
     ) -> None:
         super().__init__()
         self.project = project
         self.jobset = jobset
         self.eval_num = eval_num
+        self.absolute_eval_num = absolute_eval_num
         self.mapping = mapping
         self.hydra_clock_names = hydra_clock_names
 
+    def _get_builds(self, eval_num: int, absolute_eval_num: bool, params: str = "") -> List[int]:
+        """
+        Recursive function that returns a list of build numbers given an eval_num
+        and whether it is an absolute eval num.
+
+        Parameters
+        ----------
+        eval_num : int
+            An integer that specifies the evaluation to download. Functionality
+            differs depending on whether `absolute_eval_num` is True
+        absolute_eval_num : bool
+            Flag that specifies if the eval_num is an absolute identifier instead of
+            a relative identifier. If True, the fetcher will find an evaluation
+            with the exact ID in `eval_num`. If False, `eval_num` should be a
+            non-negative integer with `0` being the latest evaluation and `1` being
+            the second latest evaluation, etc.
+        params : str
+            A string of query parameters used when fetching the evaluations.
+            Most commonly used for pagination. By default, "".
+
+        Returns
+        -------
+        List[int]
+            A list of builds that correspond with the eval_num
+
+        Raises
+        ------
+        ConnectionError
+            Raised if the HTTP request to get the evaluations fails.
+        IndexError
+            Raised if the relative eval_num is not valid.
+        ValueError
+            Raised if the eval_num has no associated builds.
+        """
+        resp = requests.get(
+            f"https://hydra.vtr.tools/jobset/{self.project}/{self.jobset}/evals{params}",
+            headers={"Content-Type": "application/json"},
+        )
+        if resp.status_code != 200:
+            raise ConnectionError("Unable to get evals from server.")
+        evals_json = resp.json()
+        
+        if not self.absolute_eval_num: # if relative eval_num
+            if eval_num >= len(evals_json["evals"]):
+                # check if there is a second page
+                if "next" in evals_json:
+                    return self._get_builds(
+                        eval_num - len(evals_json["evals"]),
+                        absolute_eval_num,
+                        evals_json["next"] # query param for next page
+                    )
+                else:
+                    raise IndexError(f"Invalid eval_num: {self.eval_num}")
+            self._eval_id = evals_json["evals"][eval_num]["id"]
+            return evals_json["evals"][eval_num]["builds"]
+            
+        else: # if absolute eval_num
+            self._eval_id = self.eval_num
```

Comment:
In this case, `self.eval_num` should always equal `eval_num`, so choose one (I suggest the more local `eval_num`) and use it consistently throughout the body of this function.

Reply:
This assignment is [later used](https://github.com/SymbiFlow/FPGA-Tool-Performance-Visualization-Library/blob/0c4a1b12a11650738747aff99a6c0c58e06ec3e3/ftpvl/fetchers.py#L41) for generating the output Evaluation with the absolute eval id. `self._eval_id` isn't used locally. I've renamed the variable `self._abs_eval_id` which should make things clearer.